src/components/legend/Legend.ts

```diff
@@ -5,7 +5,7 @@ import { DEFAULT_CURRENCY } from 'consts';
 
 class Legend extends Base<HTMLUListElement> {
   public constructor(store: Store) {
-    const { cart, map } = store.getOptions();
+    const { cart, map, reservedLegendLabel, hideLegendPrice } = store.getOptions();
```

Comment:
I would add breaking change and create a legend property in the options:

```
legend: {
  visible: boolean;
  priceVisible: boolean;
  reservedLabel: string;
}
```

So this would become:

```
const { cart, map, legend } = store.getOptions();
```